x/deployment/keeper/keeper.go

```diff
@@ -147,6 +147,24 @@ func (k Keeper) WithDeployments(ctx sdk.Context, fn func(types.Deployment) bool)
 	}
 }
 
+// WithDeploymentsActive filters to only those with State: Active
+func (k Keeper) WithDeploymentsActive(ctx sdk.Context, fn func(types.Deployment) bool) {
+	store := ctx.KVStore(k.skey)
+	// Question: is it possible to search by a single attribute across all Deployments
+	// with only a prefix iterator?
```

Comment:
Need to make a new index for it `(index-id,state,deployment-id)`.

If we only need to index active deployments then only active deployments need to go into that index.

Reply:
Yeah I talked with Jack a little about that, I started down that progress on another branch since it'll be more substantial. I'll remove the question since answered.